arbor/include/arbor/cable_cell_param.hpp

```diff
@@ -221,6 +225,19 @@ struct ARB_SYMBOL_VISIBLE ion_reversal_potential_method {
     mechanism_desc method;
 };
 
+template <typename TaggedMech>
+struct ARB_SYMBOL_VISIBLE scaled_mechanism {
```

Comment:
Do we need that template?

Reply:
To elaborate: 
1. Compile Times. This is a public header and also used in many places inside Arbor, so templates here can add a lot of compile time.
2. Communicating Intent. Currently, this is only ever used with DensityMechanisms, so this template gives the wrong impression of bein general.

Reply:
As discussed, I'll leave the template argument, expecting to extend this to other types of mechanisms later on.